model/src/main/java/uk/gov/hmcts/payment/api/model/Payment2Repository.java

```diff
@@ -46,4 +46,14 @@ List<Reference> findReferencesByPaymentProviderAndPaymentStatusNotInAndDateCreat
     int updatePaymentUpdatedDateTime(@Param("rollbackdate") LocalDateTime rollbackDate,
                                    @Param("ccdcasenumber") String ccdCaseNumber);
 
+    @Query(value = "SELECT p.serviceType, p.ccdCaseNumber, p.reference, f.code, p.dateCreated, p.amount, p.paymentStatus.name " +
+        "FROM Payment p " +
+        "JOIN p.paymentLink pfl " +
+        "JOIN pfl.fees f " +
+        "WHERE p.dateCreated BETWEEN :fromDate AND :toDate AND p.paymentChannel.name = :paymentChannel")
+    List<Object[]> findAllByDateCreatedBetweenAndPaymentChannel(
```

Comment:
Agreed you need to use native SQL, but can you not use TelephonyPaymentsReportDto instead of Object[] - followng the convention previous - would be something like...

Optional<List<TelephonyPaymentsReportDto>> findAllByDateCreatedBetweenAndPaymentChannel(